metaflow/plugins/retry_decorator.py

```diff
@@ -1,6 +1,22 @@
+from enum import Enum
+import os
+import signal
+import sys
+import threading
+from time import sleep
+
 from metaflow.decorators import StepDecorator
 from metaflow.exception import MetaflowException
 from metaflow.metaflow_config import MAX_ATTEMPTS
+from metaflow import current
+
+
+class RetryEvents(Enum):
+    STEP = "step"
+    PREEMPT = "instance-preemption"
+
+
+PLATFORM_EVICTED_EXITCODE = 234
```

Comment:
is this needed here - since technically we might have many different exit codes depending on the error being caught. also depending on the error backend, the exit codes can be different for the same conceptual failure mode.

Reply:
move to kubernetes / batch respectively. Kept the exit code the same for now as there shouldn't be any conflicts with the current value, but this can be changed as needed.